applications/dashboard/library/class.nestedcollection.php

```diff
@@ -533,7 +534,10 @@ private function isActive($item) {
         } else {
             $highlightRoute = url($this->highlightRoute);
         }
-        return (val('url', $item) && (trim(url(val('url', $item)), '/') == trim($highlightRoute, '/')));
+        if ($ret = ($url = $item['url'] ? true : false)) {
```

Comment:
Legibility is important. While the original code was cluttered, I think it was probably easier to determine the intent when reading it. We were verifying we had a "URL" and that it matched the "highlighted route", when formatted as a full URL. Now, there are two assignments and a ternary in a single condition of an `if` statement. It seems like a lot to unpack/understand its intent at first glance. Keep in mind that more junior developers may be working behind you.

If *I* had to refactor the same code into an easier-to-understand format, I'd probably do something similar to this:
```php
$url = $item['url'] ?? false;
if ($url) {
    $result = trim(url($url), '/')) == trim($highlightRoute, '/');
} else {
    $result = false;
}
return $result;
```

Reply:
The value of `$item['url']` should be a string. You're essentially extracting it into `$url` as a boolean. On the next line, you're calling `trim` and `url` on `$url`, which means you're passing a boolean value for string parameters.